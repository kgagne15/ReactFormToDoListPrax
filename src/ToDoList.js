import React, {useState} from "react";
import ToDo from "./ToDo";
import NewToDoForm from "./NewToDoForm";
import { v4 as uuid } from 'uuid';

const ToDoList = () => {
    const INITIAL_STATE = []

    const [tasks, setTasks] = useState(INITIAL_STATE);

    const addTask = (newTask) => {
        setTasks(tasks => [...tasks, {...newTask, id: uuid()}])
    }

    const removeTask = (id) => {
        let tasksCopy = [...tasks]
        tasksCopy.splice(id, 1);
        setTasks([...tasksCopy])
    }

    return (
        <div>
            <h3>To Do List</h3>
            <NewToDoForm addTask={addTask}/>
            <div>
                {tasks.map(({id, task}) => <ToDo removeTask={removeTask} id={id} key={id} task={task} />)}
            </div>
        </div>
    )
}

export default ToDoList;