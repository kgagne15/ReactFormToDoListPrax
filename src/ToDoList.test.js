import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ToDoList from "./ToDoList";

it("renders without crashing", function() {
    render(<ToDoList/>)
})

it("matches snapshot", function() {
    const {asFragment} = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
})

it("should add new task", function() {
    const toDoList = render(<ToDoList/>)
    const task = toDoList.getByLabelText("To Do:")

    expect(toDoList.queryByText("Remove Task")).not.toBeInTheDocument();

    fireEvent.change(task, { target: {value: "Laundry"}});
    const button = toDoList.getByText("Submit");
    fireEvent.click(button);

    expect(toDoList.queryByText("Remove Task")).toBeInTheDocument();
});

it("should remove a task", function() {
    const toDoList = render(<ToDoList/>)
    const task = toDoList.getByLabelText("To Do:")

    expect(toDoList.queryByText("Remove Task")).not.toBeInTheDocument();

    fireEvent.change(task, { target: {value: "Laundry"}});
    const button = toDoList.getByText("Submit");
    fireEvent.click(button);

    expect(toDoList.queryByText("Remove Task")).toBeInTheDocument();

    const removeBtn = toDoList.getByText("Remove Task");
    fireEvent.click(removeBtn);
    expect(toDoList.queryByText("Remove Task")).not.toBeInTheDocument(); 
});