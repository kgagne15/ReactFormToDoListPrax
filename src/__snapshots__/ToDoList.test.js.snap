// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      To Do List
    </h3>
    <form>
      <label
        for="task"
      >
        To Do: 
      </label>
      <input
        id="task"
        name="task"
        placeholder="To Do"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
